be/src/exec/hdfs_scanner.cpp

```diff
@@ -386,14 +386,14 @@ void HdfsScannerContext::update_materialized_columns(const std::unordered_set<st
 void HdfsScannerContext::append_or_update_not_existed_columns_to_chunk(ChunkPtr* chunk, size_t row_count) {
     if (not_existed_slots.empty() || row_count < 0) return;
     ChunkPtr& ck = (*chunk);
+    ck->set_num_rows(row_count);
```

Comment:
I can't understand why the UT is fixed by moving this statement here.

Reply:
because here in `chunk::check_or_die`, we will check if all columns have same size, everytime we append / update a column into this chunk. 

```
void Chunk::append_or_update_column(ColumnPtr column, SlotId slot_id) {
    if (_slot_id_to_index.contains(slot_id)) {
        _columns[_slot_id_to_index[slot_id]] = std::move(column);
    } else {
        _slot_id_to_index[slot_id] = _columns.size();
        _columns.emplace_back(std::move(column));
    }
    check_or_die();
}

void Chunk::check_or_die() {
    if (_columns.empty()) {
        CHECK(_schema == nullptr || _schema->fields().empty());
        CHECK(_cid_to_index.empty());
        CHECK(_slot_id_to_index.empty());
    } else {
        for (const ColumnPtr& c : _columns) {
            CHECK_EQ(num_rows(), c->size());
            c->check_or_die();
        }
    }
}

```

Howeve in many test cases(or even prod case), we create chunk before calling `get_next`, and assign a default size.

Following code is from unit test.

```

ChunkPtr HdfsScanNodeTest::_create_chunk() {
    ChunkPtr chunk = std::make_shared<Chunk>();

    auto col1 = ColumnHelper::create_column(TypeDescriptor::from_logical_type(LogicalType::TYPE_INT), true);
    auto col2 = ColumnHelper::create_column(TypeDescriptor::from_logical_type(LogicalType::TYPE_BIGINT), true);
    auto col3 = ColumnHelper::create_column(TypeDescriptor::from_logical_type(LogicalType::TYPE_VARCHAR), true);
    auto col4 = ColumnHelper::create_column(TypeDescriptor::from_logical_type(LogicalType::TYPE_DATETIME), true);
    chunk->append_column(col1, 0);
    chunk->append_column(col2, 1);
    chunk->append_column(col3, 2);
    chunk->append_column(col4, 3);

    return chunk;
}```

Following code is from production code

```
Status HiveDataSource::get_next(RuntimeState* state, ChunkPtr* chunk) {
    if (_no_data) {
        return Status::EndOfFile("no data");
    }
    _init_chunk(chunk, _runtime_state->chunk_size());
    do {
        RETURN_IF_ERROR(_scanner->get_next(state, chunk));
    } while ((*chunk)->num_rows() == 0);
...
}
```

So when during append/update column,  there wil be inconsistent state during this operation. But at last, we will guarantee all columns have the same size.